src/pages/map.tsx

```diff
@@ -112,17 +135,45 @@ const MapPage = ({ schedule }: Props) => {
     lokasi: item
   }));
 
+  const setInitialMapBound = (mapBox: MapboxGl.Map) => {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    if ('permissions' in navigator) {
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      window.navigator.permissions.query({ name: 'geolocation' }).then(data => {
+        const permission = data.state === 'granted';
+        setGetGeoPermission(permission);
+      });
+    }
+  };
+
+  React.useEffect(() => {
+    const listOfCoordinate = coordinates.map(data => ({
+      lat: Number(data.lat),
+      lng: Number(data.lng)
+    }));
+    if (map) {
+      if (geoObj?.lat && geoObj.lng) {
+        listOfCoordinate.push({ lat: geoObj.lat, lng: geoObj.lng });
+        map.fitBounds(getMapBounds(listOfCoordinate), { padding: 100 });
+      } else {
+        map.fitBounds(getMapBounds(coordinates as LngLat[]), { padding: 100 });
+      }
+    }
+  }, [coordinates, geoObj?.lat, geoObj?.lng, map]);
+
+  const jakartaLatLng = [106.836715, -6.163088] as [number, number];
   return (
     <Container minHeight="100vh">
       <Map
+        center={jakartaLatLng}
```

Comment:
I think need to handle the case when using controlled props with state. This will always zoom to bound when rerender.

https://user-images.githubusercontent.com/856609/124578859-2d793e00-de81-11eb-94d8-19a9859b1e7d.mp4

Reply:
Should be fixed by now.. previously the fitBound always trigger on click due to the wrong useEffect deps

https://user-images.githubusercontent.com/5230095/124584060-02411f80-de7e-11eb-9051-7f88955845ae.mp4

Reply:
awesome 👍